### PR TITLE
chore: Update Typescript to 4.4.4

### PR DESCRIPTION
**Pull request by lipis**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": ["@excalidraw/eslint-config", "react-app"],
   "rules": {
-    "import/no-anonymous-default-export": "off"
+    "import/no-anonymous-default-export": "off",
+    "no-restricted-globals": "off"
```
Comment (lipis) [inline .eslintrc.json:5]: I'll add those to our configs

File: package.json [hunk 2]
```diff
@@ -72,6 +72,9 @@
     "prettier": "2.4.1",
     "rewire": "5.0.0"
   },
+  "resolutions": {
+    "@typescript-eslint/typescript-estree": "5.3.0"
+  },
```
Comment (lipis) [inline package.json:77]: taken from here: https://github.com/facebook/create-react-app/issues/7255

Comment (lipis) [reply package.json:77]: This until CRA will update it's own..

Comment (dwelle) [reply package.json:77]: yep

**Review by dwelle — APPROVED** [review]

**Comment by dwelle** [comment]
installed `@types/web` in [`836c5ec` (#4188)](https://github.com/excalidraw/excalidraw/pull/4188/commits/836c5ecd84385f6ce326ebbe5560b140dad74e87) because TS 4.4 has a regression in DOM types (in particular the `hashchange` event handler event type) which was fixed in latest but not backported to TS 4.4.

**Pull request merged**